samplers/src/main/java/io/opentelemetry/contrib/sampler/LinksBasedSampler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.sampler;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A Sampler that uses the sampled flag of the span links if present. If at least one span link is
+ * sampled, then this span will be sampled. Otherwise, it is not sampled. If the span has no span
+ * links, this Sampler will use the "root" sampler that it is built with.
+ */
+@Immutable
+final class LinksBasedSampler implements Sampler {
+
+  private final Sampler root;
+
+  private static final SamplingResult POSITIVE_SAMPLING_RESULT = SamplingResult.recordAndSample();
+  private static final SamplingResult NEGATIVE_SAMPLING_RESULT = SamplingResult.drop();
+
+  LinksBasedSampler(Sampler root) {
```

Comment:
The class and constructor are package private so it doesn't appear this can actually be used.

Reply:
Good point! Updated.